feed-formats/buttwoo.js

```diff
@@ -0,0 +1,361 @@
+// SPDX-FileCopyrightText: 2022 Andre 'Staltz' Medeiros
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+const bipf = require('bipf')
+const bfe = require('ssb-bfe')
+const blake3 = require('blake3')
+const SSBURI = require('ssb-uri2')
+const ssbKeys = require('ssb-keys')
+
+function base64ToBuffer(str) {
+  var i = str.indexOf('.')
+  return Buffer.from(str.substring(0, i), 'base64')
+}
+
+const BUTTWOO_FEED_TF = bfe.toTF('feed', 'buttwoo-v1')
+
+module.exports = function init(ssb) {
+  const feedFormat = {
+    name: 'buttwoo',
+    encodings: ['js', 'bipf'],
+
+    getFeedId(nativeMsg) {
+      const pAuthor = 6 // we can assume this because the feed format is fixed
```

Comment:
this doesn't work because length is stored as a varint, so for larger messages this is not 6 ;-)

Reply:
Damn. I thought I had a smart idea here :smiling_face_with_tear: